stew/ptrops.nim

```diff
@@ -0,0 +1,70 @@
+# stew
+# Copyright 2018-2019 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+# Pointer operations and helpers - generally, dereferencing pointers that have
+# been offset is unsafe and many of the operations herein have undefined
+# and platform-specific behavior in corner cases.
+
+# Due to poor codegen and subsequent lack of inlining, many of these operations
+# are templates even where they could be func.
+
+# ByteAddress in std lib is signed - this leads to issues with overflow checking
+# when address is on boundary.
+type
+  MemAddress* = distinct uint
```

Comment:
What is the semantic difference between a `pointer` and a `MemAddress`? I would say they are the same and this module could be simplified to avoid converting back-and-forth between them.

Reply:
you're right, as is it doesn't make much sense. it's a bit of a stepping stone to fix `ByteAddress` which is really broken, but we could remove it - or as an alternative, remove `offset` and friends overloads for `pointer`.

Reply:
removed